packages/components/src/checkbox/src/Checkbox.tsx

```diff
@@ -0,0 +1,128 @@
+import { CheckboxContext } from "./CheckboxContext.ts";
+import { useContextProps, Checkbox as RACCheckbox, type CheckboxProps as RACCheckboxProps, composeRenderProps } from "react-aria-components";
+import { cssModule } from "../../utils/src/cssModule.ts";
+import { composeClassnameRenderProps, SlotProvider } from "../../utils/index.ts";
+import { Text, TextContext } from "../../Text/index.ts";
+import { isTextOnlyChildren } from "../../utils/src/isTextOnlyChildren.ts";
+import { IconContext, CheckmarkIcon, MinusIcon } from "@hopper-ui/icons";
+import { IconListContext } from "../../IconList/index.ts";
+
+import styles from "./Checkbox.module.css";
+
+export const GlobalCheckboxCssSelector = "hop-Checkbox";
+
+// Won't be needed in next react-aria-components release: https://github.com/adobe/react-spectrum/pull/5850
+const DefaultCheckboxSlot = "checkbox";
+
+export interface CheckboxProps extends StyledComponentProps<RACCheckboxProps> {
+    /**
+     * A checkbox can vary in size.
+     */
+    size?: ResponsiveProp<"sm" | "md">;
+}
+
+function Checkbox(props:CheckboxProps, ref: ForwardedRef<HTMLLabelElement>) {
+    [props, ref] = useContextProps({ ...props, slot: props.slot || DefaultCheckboxSlot }, ref, CheckboxContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children: childrenProp,
+        size: sizeProp = "md",
+        style: styleProp,
+        ...otherProps
+    } = ownProps;
+
+    
+    const size = useResponsiveValue(sizeProp) ?? "md";
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalCheckboxCssSelector,
+        cssModule(
+            styles,
+            "hop-Checkbox",
+            size
+        ),
+        stylingProps.className
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    const children = composeRenderProps(childrenProp, prev => {
+        if (isTextOnlyChildren(prev)) {
+            return <Text>{prev}</Text>;
+        }
+
+        return prev;
+    });
+
+    return (
+        
+        <RACCheckbox
+            ref={ref}
+            className={classNames}
+            style={style}
+            {...otherProps}
+        >
+            {checkboxProps => {
+                const checkboxIconClassName = styles["hop-Checkbox__check"];
+                const icon = checkboxProps.isIndeterminate ? 
+                    <MinusIcon size="sm" className={checkboxIconClassName} /> : 
+                    <CheckmarkIcon size="sm" className={checkboxIconClassName} />;
+
+                return (
+                    <>
+                        <div className={styles["hop-Checkbox__box"]}>{icon}</div>
+                        <SlotProvider
+                            values={[
+                                [TextContext, {
```

Comment:
```
<SlotProvider
                            values={[
                                [TextContext, {
                                      className: styles["hop-Checkbox__text"],
                                      size: size
                                }],
                                [IconListContext, {
                                      className: styles["hop-Checkbox__icon-list"],
                                      size: size
                                }],
                                [IconContext, {
                                      className: styles["hop-Checkbox__icon"],
                                      size: size
                                }]
                            ]}
                        >
```

This can be simplified to this. 

In cases where IconList only fit in one slot, you don't have to target the default slot. Since there is only 1 possibility, any icon list passed will be positionned there, regardless of their slot. 

The default slot is really used when there is a slot, and a default slot assumed.
```
<Button> 
    <InfoIcon> <-- default slot 
    <Text>Allo</Text><--------------------- doesn't have to target a specific slot
    <InfoIcon slot="end-icon"> <-- Specific slot "end-icon"
</Button>
```